### DISCUSSION TITLE
The reproducibility issues of SCFAs prediction results using micom

### DISCUSSION THREAD

**Discussion by LiangenLuo** [micom-dev/micom#194] [Q&A]
Hi,

I find that every time I run 'qiime micom grow' or 'micom.workflows.grow', the resulting 'fluxes' are different, leading to varying SCFAs production for each sample.
I would like to know how to resolve this issue.

Thanks

**Comment by cdiener** [comment]
Hi. What are you using as `strategy` for the `grow` function? And which solver, database and medium? Running the same samples without changes should usually give consistent results even though fluxes don't have to be unique (algorithm is deterministic).

**Reply by LiangenLuo** [reply]
I'm sorry, I didn't describe it clearly. I'm using the sample data and example code from the q2-micom Community Tutorial.
solver:  qiime2-amplicon-2024.2-py38-linux-conda.yml
Here is my code:
qiime micom build --i-abundance cdi_table.qza \
                  --i-taxonomy cdi_taxa.qza \
                  --i-models agora103_refseq216_genus_1.qza \
                  --p-cutoff 0.0001 \
                  --p-threads 40 \
                  --o-community-models models.qza \
                  --verbose
qiime micom grow --i-models models.qza \
                 --i-medium western_diet_gut_agora.qza \
                 --p-tradeoff 0.5 \
                 --p-threads 40 \
                 --o-results growth.qza \
                 --verbose

 Problems found: 
I ran the grow model twice without changing anything, and I noticed that the results in the file 'exchange_fluxes.csv' showed different 'flux' values. Then I ran it a third time and found that the results were inconsistent with the previous two runs. such as：

running	taxon	sample_id	tolerance	reaction	flux	abundance	metabolite	direction
1	Fusobacterium	ERR1883240	1.00E-06	EX_but(e)	-28.24858731	0.095046625	but[e]	import
2	Fusobacterium	ERR1883240	1.00E-06	EX_but(e)	-12.04266508	0.095046625	but[e]	import
3	Fusobacterium	ERR1883240	1.00E-06	EX_but(e)	-19.25519755	0.095046625	but[e]	import

On the other hand, the total number of output results is also different：
running results (rows)
1 16104
2 16142
3 16119

Finally, it was found that the production of SCFAs obtained were different. such as:
running sample_id but[e] ppa[e]
1 ERR1883207 6.019178823109127 2.1169882770011683
2 ERR1883207 6.973371470007656 2.1169882770011683
2 ERR1883207 6.298594045844579 2.116988277001168
